arc/orc is broken for cpp backend using vcc (devel)

### Description

The compiler fails with an error when trying to compile using arc/orc on cpp backend any simple code such as an echo "hello". This error only appears using `--cc:vcc --cpu:i386 --threads:on`.

Possibly a continuation of this problem: #20873

### Nim Version

Nim Compiler Version 1.7.3 [Windows: amd64]
Compiled at 2022-11-29
Copyright (c) 2006-2022 by Andreas Rumpf

git hash: 555c5ed1a73fa783ee5b7526cb7a3f773f3b7f33
active boot switches: -d:release

### Current Output

```text
nim cpp -d:danger --mm:arc --cc:vcc --cpu:i386 --threads:on -f test
Hint: used config file 'E:\Nim\config\nim.cfg' [Conf]
Hint: used config file 'E:\Nim\config\config.nims' [Conf]
...................................................................
CC: E:/Nim/lib/system/exceptions.nim
CC: E:/Nim/lib/std/sysatomics.nim
CC: E:/Nim/lib/std/private/digitsutils.nim
CC: E:/Nim/lib/std/typedthreads.nim
CC: E:/Nim/lib/std/exitprocs.nim
CC: E:/Nim/lib/std/syncio.nim
CC: E:/Nim/lib/system.nim
CC: test.nim
@mE@c@sNim@slib@sstd@stypedthreads.nim.cpp
@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp
@mE@c@sNim@slib@ssystem@sexceptions.nim.cpp
@mE@c@sNim@slib@ssystem.nim.cpp
@mtest.nim.cpp
@mE@c@sNim@slib@sstd@sexitprocs.nim.cpp
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(57): error C2664: 'long _InterlockedCompareExchange(volatile long *,long,long)': não é possível converter um argumento 1 de 'volatile NI *' em 'volatile long *'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(57): note: Tipos apontados não são relacionados; conversão requer reinterpret_cast, conversão C-style ou conversão function-style entre parênteses
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\intrin0.inl.h(149): note: consulte a declaração de '_InterlockedCompareExchange'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(57): note: ao tentar corresponder a lista de argumentos '(volatile NI *, NI32, NI32)'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(75): error C2664: 'long _InterlockedCompareExchange(volatile long *,long,long)': não é possível converter um argumento 1 de 'volatile NI *' em 'volatile long *'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(75): note: Tipos apontados não são relacionados; conversão requer reinterpret_cast, conversão C-style ou conversão function-style entre parênteses
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\intrin0.inl.h(149): note: consulte a declaração de '_InterlockedCompareExchange'
@mE@c@sNim@slib@sstd@sprivate@sdigitsutils.nim.cpp
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(75): note: ao tentar corresponder a lista de argumentos '(volatile NI *, NI32, NI32)'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(84): error C2664: 'long _InterlockedExchange(volatile long *,long)': não é possível converter um argumento 1 de 'volatile NI *' em 'volatile long *'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(84): note: Tipos apontados não são relacionados; conversão requer reinterpret_cast, conversão C-style ou conversão function-style entre parênteses
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\intrin0.inl.h(175): note: consulte a declaração de '_InterlockedExchange'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(84): note: ao tentar corresponder a lista de argumentos '(volatile NI *, NI32)'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(93): error C2664: 'long _InterlockedExchange(volatile long *,long)': não é possível converter um argumento 1 de 'volatile NI *' em 'volatile long *'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(93): note: Tipos apontados não são relacionados; conversão requer reinterpret_cast, conversão C-style ou conversão function-style entre parênteses
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\intrin0.inl.h(175): note: consulte a declaração de '_InterlockedExchange'
E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp(93): note: ao tentar corresponder a lista de argumentos '(volatile NI *, NI32)'
@mE@c@sNim@slib@sstd@ssyncio.nim.cpp
Error: execution of an external compiler program 'vccexe.exe /c --platform:x86  /nologo /EHsc /O2   /IE:\Nim\lib /IC:\Users\josep\desktop /nologo /FoE:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp.obj E:\nimcache\test_r\@mE@c@sNim@slib@sstd@ssysatomics.nim.cpp' failed with exit code: 2
```


### Expected Output

```text
Compile without error
```


### Possible Solution

Change static_cast from NI to long

### Additional Information

_No response_

--- a/lib/std/sysatomics.nim
+++ b/lib/std/sysatomics.nim
@@ -190,12 +190,12 @@ elif someVcc:
     proc interlockedCompareExchange64(p: pointer; exchange, comparand: int64): int64
       {.importcpp: "_InterlockedCompareExchange64(static_cast<NI64 volatile *>(#), #, #)", header: "<intrin.h>".}
     proc interlockedCompareExchange32(p: pointer; exchange, comparand: int32): int32
-      {.importcpp: "_InterlockedCompareExchange(static_cast<NI volatile *>(#), #, #)", header: "<intrin.h>".}
+      {.importcpp: "_InterlockedCompareExchange(static_cast<long volatile *>(#), #, #)", header: "<intrin.h>".}
     proc interlockedCompareExchange8(p: pointer; exchange, comparand: byte): byte
       {.importcpp: "_InterlockedCompareExchange8(static_cast<char volatile *>(#), #, #)", header: "<intrin.h>".}
     proc interlockedExchange8(location: pointer; desired: int8): int8 {.importcpp: "_InterlockedExchange8(static_cast<NI8 volatile *>(#), #)", header: "<intrin.h>".}
     proc interlockedExchange16(location: pointer; desired: int16): int16 {.importcpp: "_InterlockedExchange16(static_cast<NI16 volatile *>(#), #)", header: "<intrin.h>".}
-    proc interlockedExchange32(location: pointer; desired: int32): int32 {.importcpp: "_InterlockedExchange(static_cast<NI volatile *>(#), #)", header: "<intrin.h>".}
+    proc interlockedExchange32(location: pointer; desired: int32): int32 {.importcpp: "_InterlockedExchange(static_cast<long volatile *>(#), #)", header: "<intrin.h>".}
     proc interlockedExchange64(location: pointer; desired: int64): int64 {.importcpp: "_InterlockedExchange64(static_cast<NI64 volatile *>(#), #)", header: "<intrin.h>".}
   else:
     proc interlockedCompareExchange64(p: pointer; exchange, comparand: int64): int64

